lib/sparrow/h2_worker.ex

```diff
@@ -78,8 +78,16 @@ defmodule Sparrow.H2Worker do
   end
 
   def handle_info(:ping, state) do
-    H2Adapter.ping(state.connection_ref)
-    _ = schedule_message_after(:ping, state.config.ping_interval)
+    _ =
+      case state.config.ping_interval do
```

Comment:
I would consider using `and` (instead of this case) here as we want to do sth only when ping_interval is "truish" and we don't care about the result.